src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -32,23 +32,25 @@ private static bool IsWindows7()
             return version.Major == 6 && version.Minor == 1;
         }
 
+        private static bool UseFileAPIs(bool isRedirected) => isRedirected || Console.InputEncoding.CodePage != UnicodeCodePage;
```

Comment:
This is a change in behavior from the current code, where OutputEncoding rather than InputEncoding is checked for OpenStandardOutput/OpenStandardError.

Reply:
Great catch #solved